gradle/ide.gradle

```
@@ -0,0 +1,141 @@
+import org.jetbrains.gradle.ext.Remote
+import org.jetbrains.gradle.ext.JUnit
+
+buildscript {
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+  dependencies {
+    classpath "gradle.plugin.org.jetbrains.gradle.plugin.idea-ext:gradle-idea-ext:0.7"
+  }
+}
+
+apply plugin: org.jetbrains.gradle.ext.IdeaExtPlugin
+
+allprojects {
+  apply plugin: 'idea'
+
+  tasks.named('idea') {
+    doFirst { throw new GradleException("Use of the 'idea' task has been deprecated. For details on importing into IntelliJ see CONTRIBUTING.md.") }
+  }
+}
+
+tasks.register('configureIdeaGradleJvm') {
+  group = 'ide'
+  description = 'Configures the appropriate JVM for Gradle'
+
+  doLast {
+    modifyXml('.idea/gradle.xml') { xml ->
+      def gradleSettings = xml.component.find { it.'@name' == 'GradleSettings' }.option[0].GradleProjectSettings
+      // Remove configured JVM option to force IntelliJ to use the project JDK for Gradle
+      gradleSettings.option.findAll { it.'@name' == 'gradleJvm' }.each { it.parent().remove(it) }
+    }
+  }
+}
+
+tasks.register('configureIdeaRunConfigs') {
+  group = 'ide'
+  description = 'Configures default run configuration settings'
+
+  doLast {
+    modifyXml('.idea/workspace.xml') { xml ->
+      def runManager = xml.component.find { it.'@name' == 'RunManager' }
+      if (runManager == null) {
+        throw new GradleException("IntelliJ 'RunManager' configuration is missing from workspace.xml. You may need to refresh your Gradle project.")
+      }
+
+      def debugConfig = runManager.configuration.find { it.'@name' == 'Debug Elasticsearch' }
+      // Enable "auto restart" on remote debug run configuration
+      if (debugConfig.option.any { it.'@name' == 'AUTO_RESTART' && it.'@value' == 'true'} == false) {
+        def restart = new NodeBuilder().option(name: 'AUTO_RESTART', value: 'true')
+        debugConfig.append(restart)
+      }
+    }
+  }
+}
+
+idea {
+  project {
+    vcs = 'Git'
+    jdkName = '13'
+
+    settings {
+      delegateActions {
+        delegateBuildRunToGradle = false
+        testRunner = 'choose_per_test'
+      }
+      taskTriggers {
+        afterSync tasks.named('configureIdeaGradleJvm'), tasks.named('configureIdeaRunConfigs')
+      }
+      codeStyle {
+        java {
+          classCountToUseImportOnDemand = 99
```

Is this number sufficiently absurdly high?
```suggestion
          classCountToUseImportOnDemand = 9999
```

---

I've never encountered a scenario in which IntelliJ used a wildcard import in all the time I've set this to 99. We can certainly increase it though, I used 99 just because the UI textbox fits 2 characters 😄